handler/validation/openapi.go

```diff
@@ -28,10 +32,63 @@ func NewOpenAPI(opts *OpenAPIOptions) *OpenAPI {
 	}
 }
 
-func (v *OpenAPI) ValidateRequest(req *http.Request) error {
-	route, pathParams, err := v.options.router.FindRoute(req.Method, req.URL)
+func (v *OpenAPI) getModifiedSwagger(key, origin string) (*openapi3.Swagger, error) {
+	swagger, exists := swaggers.Load(key)
+	if !exists {
+		clonedSwagger := cloneSwagger(v.options.swagger)
+
+		var newServers []string
+		for _, s := range clonedSwagger.Servers {
```

Comment:
What about an empty server list or a list which does not match the given origin.
This case still fails:
```yaml
openapi: '3'
info:
  title: 'Couper backend validation test: non-canonical server URL'
  version: 'v1.2.3'
servers:
#  - url: https://httpbin.org/
  - url: http://httpbin.org/
paths:
  /get:
    get:
      responses:
        200:
          description: OK
```

```hcl
server "couper" {
  endpoint "/" {
	proxy {
      backend {
        origin = "https://httpbin.org"
        path = "/get"
        openapi {
          file = "schema.yaml"
        }
      }
	}
  }
}

```